File: 2023.1/TEP/lista_3/A.py
import math

a, b, c = map(int, input().split())

lst = []

if a == 0:
    print(-1)
else:
    delta = b**2 - 4*a*c
    if delta < 0:
        print(0)
    else:
        x1 = (-b + math.sqrt(delta)) / (2*a)
        x2 = (-b - math.sqrt(delta)) / (2*a)
        if isinstance(x1, (int, float)):
            lst.append(x1)
        if isinstance(x2, (int, float)):
            lst.append(x2)
        lst.sort()
        print(len(lst))
        for i in lst:
            print(f"{i:.10f}")

File: 2023.1/TEP/lista_3/A.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

int main(){

    int a, b, c, delta;
    double x1, x2;

    scanf("%d%d%d", &a, &b, &c);

    if(a == 0)
        printf("-1\n");

    else{
        delta = (b*b) - (4*a*c);
        
        if(delta < 0)
            printf("0\n");

        else{
            x1 = (-b + sqrt(delta))/2*a;
            x2 = (-b - sqrt(delta))/2*a;

            if(x1 == x2){
                printf("1\n");
                printf("%.10lf\n", x1);
            }
            
            else{
                printf("2\n");
                if(x1 < x2)
                    printf("%.10lf\n%.5lf\n", x1, x2);
                else
                    printf("%.10lf\n%.5lf\n", x2, x1);
            }
        }
    }

    return 0;
}